libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1064,19 +1068,23 @@ string YulUtilFunctions::clearStorageRangeFunction(Type const& _type)
 {
 	string functionName = "clear_storage_range_" + _type.identifier();
 
-	solAssert(_type.storageBytes() >= 32, "Expected smaller value for storage bytes");
-
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
-			function <functionName>(start, end) {
+			function <functionName>(start, offset, end) {
+				if gt(offset, 0) {
+					let mask := <shr>(mul(8, sub(32, offset)), <ones>)
+					sstore(start, and(mask, sload(start)))
+				}
```

Comment:
Why is this done only for `start`?

Reply:
Wait, what is `offset` pointing to exactly?

Reply:
If types are packed in storage, `offset` points to the first part of `slot` that needs to be cleared. For example, we have `uint32[]` storage array and we want to resize it to  `1`. Here offset would be `4` as we want to clear only 4 bytes of the first slot where array data is stored.

This happens in cases when we are overwriting arrays in storage (https://github.com/ethereum/solidity/pull/9738#issuecomment-687206828). So we do not want to clear whole array, just some part of it. This logic is done only for first slot, as it is the only slot that will be partially cleared.

Reply:
Ok, I got the first part, but I still don't understand why only the first slot. What if you're resizing it to 6 elements, you need 2 slots and the offset would be 8 of the second slot?

Reply:
This function was intended to clear all slots from `start` until `end`. For arrays it was called with `start` and `end` considering new and old length. So, `start` will point to first slot we actually want to clear. And that is the only slot we want to partially clear. All other slots are set to zero until the end. 
Hope it was the question you asked and it clears what is being done here :)

Reply:
Ahhh ok